Fix: Responsive UI for search bar on HomePage

mithun50 opened the pull request:
Fix search bar overflow on mobile devices
Fixes : #72 
The search bar in the 'Available Courses' section was overflowing the viewport on mobile devices, causing horizontal scrolling.

This commit fixes the issue by applying responsive styles to the container of the search bar and title. The container now stacks its children vertically on small screens and horizontally on larger screens, preventing the overflow. The search input is also set to full width on mobile for better usability.

mithun50 committed:
Fix: Responsive UI for search bar on HomePage

- Updated layout in src/pages/HomePage.tsx
- Fixed overflow issue on small screens
- Stacks vertically on mobile, horizontal on larger screens

Co-authored-by: mithun50 <mithungowda.b7411@gmail.com>

mithun50 commented:
@chandansgowda review required

chandansgowda reviewed (changes requested):
Search icon is moving to the next line (below available course). Its not looking good on mobile.. rather can we just have available courses title in the center and below that have the expanded search (only in mobile view). What do you think?

mithun50 commented:
@chandansgowda I think it's quite good 
Fixed

mithun50 committed:
Update HomePage.tsx

Search icon is moving to the next line (below available course). Its not looking good on mobile.. rather can we just have available courses title in the center and below that have the expanded search (only in mobile view).
